content/blog/open-source-monitoring-at-the-hpe-customer-technology-center-böblingen.md

```diff
@@ -0,0 +1,377 @@
+
+A directory of the available Redfish® resources at an interface can be retrieved with the following RESTful API command: 
+
+\`GET {{baseUrl}}/redfish/v1/resourcedirectory\`
+
+I used [Postman](https://www.postman.com/) as an developer tool to retrieve and search the resource directory. The above RESTful API comand was used in Postman with a basic username/password authorization at one of the ILOs in the CTC, where the baseUrl is defined as <https://iloIPaddress>. I searched in the obtained resource directory for "thermal" in order to find the correct command to retrieve thermal data:
+
+![](/img/redfishresourcedirectory.png "Redfish resource directory (Postman output)")
+
+Hence, I can retrieve thermal server data with the following RESTful command:
+
+\`GET /redfish/v1/chassis/1/thermal\`
+
+And using this Get command with Postman confirmed that I do get the values of all temperature sensors of the server:
+
+![](/img/redfishtemparture.png "ILO Redfish thermal information (Postman output)")
+
+# Building the HPE Server Exporter
+
+Great, now that I do know the RESTful command to retrieve the data I am looking for and I do have a Prometheus-Grafana based monitoring established in the Customer Technology Center, I was only missing a Prometheus exporter for the ILO interfaces and the according Grafana dashboards. Hence I needed to write another Prometheus exporter: one for the ILO interface. I decided to use Python as the programming language for this connector, because I could build on the knowledge gained when I developed the SimpliVity-Prometheus exporter. 
+
+First I had to decide which redfish library I want to use: the [DMTF Redfish Python library](https://github.com/DMTF/python-redfish-library) or the [HPE Python Redfish library](https://github.com/HewlettPackard/python-ilorest-library) - they cannot coexist in a Python environment. I decided to use the DMTF Redfish Python library, since I didn't need the additional features of the HPE Python Redfish library for my use case. The retrieved data can easily made digestible for Prometheus with the prometheus_client Python library. The additional libraries, that I used are mainly for providing the input data in XML-format (etree library) and for the permanent monitoring loop (time, sys and os library) and logging (datetime library). Hence, I needed the following libraries included in my Python script:
```

Comment:
Modify the first sentence to:
"First, I had to decide on which Redfish library I wanted to use - the DMTF Redfish Python library or the HPE Python Redfish library - as they cannot both coexist in a Python environment."